Automattic/sensei
# Fix/issue 2885: Sync questions in quiz if user has not taken any action.

Fixes #2885 

#### Changes proposed in this Pull Request:

* This PR checks, every time a quiz is loaded, if the user has taken any action yet (saved or completed the quiz) or not. In case the user hasn't (the user has only viewed the quiz), then the commentmeta meta_key `questions_asked` is set to empty. 
This causes the following behavior: every time a user (that has not taken any action) loads a quiz, the questions will be fetched from the posts table, not from the `questions_asked` commentmeta meta_key. This is done by the `lesson_quiz_questions` function. It basically makes the process that ran only the first time a user saw a quiz, run every time while the user hasn't taken any action.

#### Testing instructions:

TESTING A QUIZ THAT SHOWS ALL QUESTIONS
* Create a course with a lesson with some questions (a lesson quiz).
* Set the quiz to display All questions in the Quiz Settings metabox.
* Sign in with a learner to that lesson and go to the quiz.
* Don't do anything (don't save it nor complete it). Just "view" the quiz.
* Add or delete, from the admin dashboard, a question in that same quiz.
* Go back to the quiz with that learner that only "saw" the quiz (refresh the page if needed), you should see the updated questions. (The new ones added should appear. The deleted ones should not appear).
 
TESTING A QUIZ THAT SHOWS  A SPECIFIC NUMBER OF RANDOM QUESTIONS
* Create a course with a lesson with some questions (a lesson quiz).
* Set the quiz to display "a number" of questions in the Quiz Settings metabox. (This number should be lower than the total number of questions).
* Sign in with a learner to that lesson and go to the quiz.
* Don't do anything (don't save it nor complete it). Just "view" the quiz.
* Add or delete, from the admin dashboard, a question in that same quiz.
* Go back to the quiz with that learner that only "saw" the quiz, you should see the updated questions taken into consideration for the random selection(refresh the page if needed). (The new ones added should appear. The deleted ones should not appear). You might need to refresh a few times (depending on how many questions you have) so that you can see that the random questions selection now include the new ones and don't include the deleted ones.



* Note: If a user saved or completed a quiz, but resets it (if this is enabled), the behavior is as if the user has never taken action (only seen the quiz).
